rust/src/detect.rs

```diff
@@ -0,0 +1,312 @@
+    let (i, mode) = detect_parse_uint_mode(i)?;
+    let (i, _) = opt(is_a(" "))(i)?;
+    let (i, arg1) = map_opt(digit1, |s: &str| s.parse::<T>().ok())(i)?;
+
+    match mode {
+        DetectUintMode::DetectUintModeNe => {}
+        DetectUintMode::DetectUintModeLt => {
+            if arg1 == T::min_value() {
+                return Err(Err::Error(make_error(i, ErrorKind::Verify)));
+            }
+        }
+        DetectUintMode::DetectUintModeLte => {
+            if arg1 == T::max_value() {
+                return Err(Err::Error(make_error(i, ErrorKind::Verify)));
+            }
+        }
+        DetectUintMode::DetectUintModeGt => {
+            if arg1 == T::max_value() {
+                return Err(Err::Error(make_error(i, ErrorKind::Verify)));
+            }
+        }
+        DetectUintMode::DetectUintModeGte => {
+            if arg1 == T::min_value() {
+                return Err(Err::Error(make_error(i, ErrorKind::Verify)));
+            }
+        }
+        _ => {
+            return Err(Err::Error(make_error(i, ErrorKind::MapOpt)));
+        }
+    }
+
+    Ok((
+        i,
+        DetectUintData {
+            arg1,
+            arg2: T::min_value(),
+            mode: mode,
+        },
+    ))
+}
+
+pub fn detect_match_uint<T: DetectIntType>(x: &DetectUintData<T>, val: T) -> bool {
+    match x.mode {
+        DetectUintMode::DetectUintModeEqual => {
+            if val == x.arg1 {
+                return true;
+            }
+        }
+        DetectUintMode::DetectUintModeNe => {
+            if val != x.arg1 {
+                return true;
+            }
+        }
+        DetectUintMode::DetectUintModeLt => {
+            if val < x.arg1 {
+                return true;
+            }
+        }
+        DetectUintMode::DetectUintModeLte => {
+            if val <= x.arg1 {
+                return true;
+            }
+        }
+        DetectUintMode::DetectUintModeGt => {
+            if val > x.arg1 {
+                return true;
+            }
+        }
+        DetectUintMode::DetectUintModeGte => {
+            if val >= x.arg1 {
+                return true;
+            }
+        }
+        DetectUintMode::DetectUintModeRange => {
+            if val > x.arg1 && val < x.arg2 {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+pub fn detect_parse_uint<T: DetectIntType>(i: &str) -> IResult<&str, DetectUintData<T>> {
+    let (i, _) = opt(is_a(" "))(i)?;
+    let (i, uint) = alt((
+        detect_parse_uint_start_interval,
+        detect_parse_uint_start_equal,
+        detect_parse_uint_start_symbol,
+    ))(i)?;
+    let (i, _) = all_consuming(take_while(|c| c == ' '))(i)?;
+    Ok((i, uint))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_detect_u64_parse(
+    str: *const std::os::raw::c_char,
+) -> *mut DetectUintData<u64> {
+    let ft_name: &CStr = CStr::from_ptr(str); //unsafe
```

Comment:
cc @jasonish 
I think it is unsafe, because you have no assurance that C caller will provide a valid null-terminated string...

Changing `str` variable name

Reply:
> if unsafe, should we do more to check? E.g. can `ft_name` be validated somehow, or can `str` be?

We can check for `null`, thats about it.